src/opt/test/test_renderer.cc

```diff
@@ -122,43 +142,51 @@ void TestRendererPixelAccuracy(const Camera& camera) {
   cv::Mat_<cv::Vec3b> color_image(camera.height(), camera.width());
   renderer->DownloadColorResult(camera.width(), camera.height(), color_image.data);
   
-//   // DEBUG: display.
-//   cv::Mat_<float> depth_mat(camera.height(), camera.width(), depth_image.get());
-//   cv::imshow("rendered depth", depth_mat);
-//   cv::imshow("rendered color", color_image);
-//   cv::waitKey(0);
+  // DEBUG: display.
+  cv::Mat_<float> depth_mat(camera.height(), camera.width(), depth_image.get());
+  cv::imshow("rendered depth", depth_mat * (20.f/255.f));
+  cv::imshow("rendered color", color_image);
+  cv::waitKey(0);
   
   // Compare rendered images to the mesh.
   // NOTE: Due to placing vertices exactly at pixel positions and the fill rule
   // used in rendering, the right- and bottom-most pixels will not be covered by
   // the mesh. Thus the test checks that they are still at their original value.
-  const float* depth_ptr = depth_image.get();
   std::size_t i = 0;
-  for (int y = 0; y < camera.height(); ++ y) {
-    for (int x = 0; x < camera.width(); ++ x) {
+  for (int y = 0; y < camera.height(); y += kStep) {
+    for (int x = 0; x < camera.width(); x += kStep) {
       cv::Vec3b rgb_color = color_image(y, x);
-      if (y < camera.height() - 1 && x < camera.width() - 1) {
-        ASSERT_NEAR(*depth_ptr, mesh_vertex_cloud[i].z, 1e-3f)
-            << "Error at pixel (" << x << ", " << y << ")";
-        ASSERT_EQ(rgb_color[0], mesh_vertex_cloud[i].r)
-            << "Error at pixel (" << x << ", " << y << ")";
-        ASSERT_EQ(rgb_color[1], mesh_vertex_cloud[i].g)
-            << "Error at pixel (" << x << ", " << y << ")";
-        ASSERT_EQ(rgb_color[2], mesh_vertex_cloud[i].b)
-            << "Error at pixel (" << x << ", " << y << ")";
-      } else {
-        ASSERT_EQ(*depth_ptr, 0)
-            << "Error at pixel (" << x << ", " << y << ")";
-        ASSERT_EQ(rgb_color[0], 0)
-            << "Error at pixel (" << x << ", " << y << ")";
-        ASSERT_EQ(rgb_color[1], 0)
-            << "Error at pixel (" << x << ", " << y << ")";
-        ASSERT_EQ(rgb_color[2], 0)
-            << "Error at pixel (" << x << ", " << y << ")";
+      float depth_value = depth_mat(y, x);
+      pcl::PointXYZRGB point = mesh_vertex_cloud[i];
+      Eigen::Vector3f p = point.getVector3fMap();
+      Eigen::Vector2f image_point = camera.ProjectToImageCoordinates(Eigen::Vector2f(p.x()/p.z(), p.y()/p.z()));
+      if(!isinf(image_point.squaredNorm())){
+        if (depth_value > 0) {
```

Comment:
Could the change in the condition here hide errors where the renderer incorrectly returns zero depth and color for all pixels?